### PR TITLE
feat(api): Rework user details/appearance endpoints

### PR DESCRIPTION
**Pull request by billyvg**
I've merged the `user_appearance` endpoint into `user_details` and removed password and emails as fields that can be updated by this endpoint. There is a separate endpoint for adding emails and there will be a follow up PRs to:

* https://github.com/getsentry/sentry/pull/7133 - change password (will be a new endpoint but I think should be merged into the users/security endpoint introduced in #7113)
* remove user_appearance endpoint and update JSX accordingly

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Changes require @getsentry/security sign-off</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

### Security concerns found

- src/sentry/api/endpoints/sudo.py


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/api/endpoints/user_details.py [hunk 1]
```diff
@@ -60,16 +57,73 @@ class Meta:
...
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        result = serializer.object
+
+        if request.DATA.get('password'):
+            if request.DATA.get('password') != request.DATA.get('passwordVerify'):
+                return Response({"detail": "New password does not match verified password"},
+                                status=status.HTTP_400_BAD_REQUEST)
+
+            # TODO(billy): Why is `user.has_usable_password()` false here
```
Comment (macqueen) [inline src/sentry/api/endpoints/user_details.py]: usually that's if a user has never had a password set. are these sso users or something?

File: src/sentry/api/endpoints/user_details.py [hunk 2]
```diff
@@ -60,16 +57,84 @@ class Meta:
...
+        # Changing `password`, `email`, or `username` requires sudo
+        if request.DATA.get('passwordVerify') or \
+                request.DATA.get('password') or \
+                request.DATA.get('username') or \
+                request.DATA.get('email'):
+            return self.protected_put(request, user)
+        else:
+            return self.update_unprivileged_details(request, user)
+
+    def update_unprivileged_details(self, request, user):
```
[the remark this answers is not in the document]
Comment (macqueen) [reply src/sentry/api/endpoints/user_details.py]: i think this is fine

File: src/sentry/api/endpoints/user_details.py [hunk 3]
```diff
@@ -60,16 +57,84 @@ class Meta:
...
+            )
+
+        user.save()
+
+        # Proceed to update unprivileged fields
+        return self.update_unprivileged_details(request, user)
+
     def put(self, request, user):
+        # Changing `password`, `email`, or `username` requires sudo
+        if request.DATA.get('passwordVerify') or \
```
[the remark this answers is not in the document]
Comment (macqueen) [reply src/sentry/api/endpoints/user_details.py]: not really sure if this is any cleaner, but you could do something like: 

`if any(k in request.DATA for k in ['passwordVerify', 'password', 'email', 'username']):`

**Review by macqueen — APPROVED** [review]

**Comment by billyvg** [comment]
@mattrobenolt can review since there's potential security concerns here?

**Review by mattrobenolt — CHANGES_REQUESTED** [review]
@billyvg when you get some time, mind running me through this IRL just so I'm clear? I'm missing the other bits of how this is implemented to give a better review.

**Comment by billyvg** [comment]
@mattrobenolt This is ready for a review, I've merged the `user_appearance` endpoint into `user_details` and removed `password` and `emails` as fields that can be updated by this endpoint. There is a separate endpoint for adding emails and there will be a follow up PR to:
* change password (will be a new endpoint but I think should be merged into the `users/security` endpoint introduced in #7113)
* remove `user_appearance` endpoint and update JSX accordingly

**Review by mattrobenolt — APPROVED** [review]
wcgw

**Pull request merged**